controllers/controllers/resource/template.go

```diff
@@ -159,7 +166,114 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 		values["eksaVspherePassword"] = string(passwordBytes)
 	}
 
-	return generateTemplateResources(templateBuilder, clusterSpec, workloadTemplateNames, kubeadmconfigTemplateNames, cpOpt)
+	return generateTemplateResources(templateBuilder, clusterSpec, workloadTemplateNames, kubeadmconfigTemplateNames, cpOpt, nil)
+}
+
+func (r *CloudStackTemplate) TemplateResources(ctx context.Context, eksaCluster *anywherev1.Cluster, clusterSpec *cluster.Spec, csdc anywherev1.CloudStackDatacenterConfig, cpCsmc, etcdCsmc anywherev1.CloudStackMachineConfig, workerCsmcs map[string]anywherev1.CloudStackMachineConfig) ([]*unstructured.Unstructured, error) {
+	if len(workerCsmcs) > 1 {
+		return nil, fmt.Errorf("multiple worker node group configuration is not supported in CloudStack provider")
+	}
+	workerNodeGroupMachineSpecs := make(map[string]anywherev1.CloudStackMachineConfigSpec, len(workerCsmcs))
```

Comment:
i understand most of the code is copied from vsphere. but it really becomes hard to follow what this method is doing. is it possible to break it down by adding smaller methods like
- workerNodeGroupMachineSpec()
- controlPlaneTemplateName()
- workerTemplateName()
- etcdTemplateName()
- ...